dask/tests/test_distributed.py

```diff
@@ -717,23 +732,7 @@ async def test_shuffle_priority(c, s, a, b):
     ddf = dd.from_pandas(df, npartitions=10)
     ddf2 = ddf.shuffle("a", shuffle="tasks", max_branch=32)
     await c.compute(ddf2)
-
-    # Parse transition log for processing tasks
-    log = [
-        eval(l[0])[0]
-        for l in s.transition_log
-        if l[1] == "processing" and "simple-shuffle-" in l[0]
-    ]
-
-    # Make sure most "split" tasks are processing before
-    # any "combine" tasks begin
-    late_split = np.quantile(
-        [i for i, st in enumerate(log) if st.startswith("split")], 0.75
-    )
-    early_combine = np.quantile(
-        [i for i, st in enumerate(log) if st.startswith("simple")], 0.25
-    )
-    assert late_split < early_combine
```

Comment:
This test was not only flaky but also useless. The change to the shuffle split priorities that were introduced here https://github.com/dask/dask/pull/7846 are actually quite important but this test is not testing this. The time when a task transitions to processing on scheduler side is quite irrelevant since we're scheduling greedily. split tasks are **always** transitioned immediately after the grouper ended up in memory. The only case where this might not be true is if a combine task would "unlock" at the same time such that there is a race in the transition. Actual execution order is however ensured by the worker.

This test is a bit of a chicken egg situation. the test tried to capture this behavior without actually asserting that priorities is set. It still used a rather low level API. I kept the intention of not asserting on priorities but had to fiddle with the worker internals a bit. If this turns out to be unstable/hard to maintain we can reconsider. The important thing is: This test is now sensitive to the priorities set in the shuffle layer 🎉